flow/matrix_decomposition.cc

```diff
@@ -82,13 +89,16 @@ MatrixDecomposition::MatrixDecomposition(SkMatrix44 matrix) : valid_(false) {
   }
 
   scale_.fX = row[0].length();
-  row[0].normalize();
+  // TODO(garyq): use skia row[x].normalize() when skia fixes it
+  normalize(row[0]);
 
   shear_.fX = row[0].dot(row[1]);
   row[1] = SkVector3Combine(row[1], 1.0, row[0], -shear_.fX);
 
   scale_.fY = row[1].length();
-  row[1].normalize();
+  // TODO(garyq): use skia row[x].normalize() when skia fixes it
```

Comment:
You can move both TODOs in to the method implementation and add a single comment there to reminds future us to remove the workaround.